orderer/consensus/etcdraft/chain.go

```diff
@@ -1149,12 +1149,10 @@ func (c *Chain) apply(ents []raftpb.Entry) {
 				continue
 			}
 
-			if err := c.Node.storage.Sync(); err != nil {
-				c.logger.Debugf("Failed to sync raft log, error: %s", err)
-			}
+			// persist the WAL entries into disk
+			c.Node.storage.WALSyncC <- struct{}{}
```

Comment:
You need to wait for the sync, no?

Reply:
It need not to wait.  This is the reasoning for my statement, now the access to storage all moved to node.go goroutine and it is sequenced/synchronzied on select, hence, the chain apply goroutine need not to wait.

Reply:
But if we don't wait for the sync, then the goroutine of the node can be too late at syncing and in the meanwhile the goroutine of the chain will continue without the sync happening, no? That's not equivalent to the previous code where the sync is synchronous and blocking.

Reply:
as per my understanding, the sender is blocked until the receiver receives the value on the channel...
`Receivers always block until there is data to receive. If the channel is unbuffered, the sender blocks until the receiver has received the value.` [ref](https://go.dev/doc/effective_go#channels)

If this is the case, the node goroutine will get its chance to proceed further in concurrent with chain goroutine.
 
Yes, there is chance if the node goroutine is context switched between the channel receive and actual sync operation. Yes. It is possible that the chain goroutine continues its operation. 

But, still chain goroutine execution does not conflict with the execution of the node goroutine even if the current config change evicts/halts the chain.    Please correct me if I am missing anything here.

Reply:
Fine, leave this as it is.